hack/generator/pkg/codegen/pipeline_remove_type_aliases.go

```diff
@@ -63,6 +63,8 @@ func resolveTypeName(visitor *astmodel.TypeVisitor, name astmodel.TypeName, type
 		return def.Name(), nil // must remain named for controller-gen
 	case astmodel.ValidatedType:
 		return def.Name(), nil // must remain named so there is somewhere to put validations
+	case *astmodel.FlaggedType:
+		return def.Name(), nil // must remain named so there is somewhere to put flags
```

Comment:
uh, is this correct?

Reply:
They need to make it all the way through the pipeline though, so that each stage has an opportunity to consume/remove/alter them as appropriate.

Maybe I should add a check right before the output step that they're all gone? Or we could leave it as is where if a `FlaggedType` makes it through it doesn't really matter as it just delegates all of its behavior to its element.

Reply:
Maybe what's throwing you is the comment though... which yeah doesn't really make much sense now that I re-read it.

Reply:
inlining the type here shouldn't remove the fact that it is a flagged type though, it just won't be named? or does it need to be named because the pipeline stage that expects to process it doesn't recurse through other types looking for flag?

Reply:
Actually I'm not sure -- I think that this needs to return the name here because that's what we're doing for `ObjectType` too. Basically this function is saying: "When I resolve a `TypeName` and find a `FlaggedType`, then it's not a type alias so we can keep the name"

The only way something is a type alias is if when you resolve `TypeName` you find another `TypeName` (which is why that's the main case that has special logic in the switch).

Reply:
at least we should fix the comment :)

Reply:
Yeah, will fix the comment for sure - are you ok with the rest of it though (i.e. the fact these types progress through the pipeline and are treated like `ObjectType` here?)

Reply:
I don't know about @Porges, but it make sense to me! 😁

Reply:
Improved the comment, left the rest as is